### PR TITLE
Update checkboxgroup.mdx

### PR DESCRIPTION
**Pull request by SOjaHPE**
Updating the guidance associated with the recommended component usage/functionality should be used as a UI displays 0-n # of checkboxes. This change updates the guidance related to checkboxgroup. The guidance was decreased to not recommend its usage with 10 or fewer. Moved it down to 4 or fewer.

Exactly where this new content should be located in this guidance page, was torn - is the best location for this expanded guidance in **_Best Practices_** or **_The Don'ts for CheckboxGroup_**. Your feedback is welcomed to optimize this.

[Deploy Preview](https://deploy-preview-INSERT_PR_#_HERE--keen-mayer-a86c8b.netlify.app/)

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

### CODE REVIEW

File: aries-site/src/pages/components/checkboxgroup.mdx [hunk 1]
```diff
@@ -25,10 +25,11 @@ CheckBoxGroup should be used to present a group of options to the user. The user
 
 ### Best practices
 
-- CheckBoxGroup should be used to select zero or more of options from a list.
+- CheckBoxGroup should be used to select zero or more options from a list.
 - In a CheckBoxGroup, options are independent from one another.
 - Settings changed by checkboxes should not take effect until the user clicks to proceed.
-- It is recommended to have only 10 or fewer options in a CheckBoxGroup. Use multiple CheckboxGroups if necessary.
+- It is recommended to have only 4 or fewer options in a CheckBoxGroup.
+- Use [SelectMultiple](/components/SelectMultiple) when multiple selections can be made in a list that contains 5 or more items.
```
Comment (halocline) [inline aries-site/src/pages/components/checkboxgroup.mdx:32]:
```suggestion
- Use [SelectMultiple](/components/selectmultiple#best-practices) with search when multiple selections can be made in a list that contains 10 or more items.
```

File: aries-site/src/pages/components/checkboxgroup.mdx [hunk 2]
```diff
@@ -25,10 +25,11 @@ CheckBoxGroup should be used to present a group of options to the user. The user
 
 ### Best practices
 
-- CheckBoxGroup should be used to select zero or more of options from a list.
+- CheckBoxGroup should be used to select zero or more options from a list.
 - In a CheckBoxGroup, options are independent from one another.
 - Settings changed by checkboxes should not take effect until the user clicks to proceed.
-- It is recommended to have only 10 or fewer options in a CheckBoxGroup. Use multiple CheckboxGroups if necessary.
+- It is recommended to have only 4 or fewer options in a CheckBoxGroup.
+- Use [SelectMultiple](/components/selectmultiple#best-practices) with search when multiple selections can be made in a list that contains 10 or more items.
```
Comment (taysea) [inline aries-site/src/pages/components/checkboxgroup.mdx:32]: Feels like we should also have the bullet here about using SelectMultiple when there are >4 but <10 items.

Comment (halocline) [reply aries-site/src/pages/components/checkboxgroup.mdx:32]:
```suggestion
- Use [SelectMultiple](/components/selectmultiple#best-practices) with search when multiple selections can be made in a list that contains 5-9 items; use [SelectMultiple](/components/selectmultiple#best-practices) with search when containing 10 or more items.
```

**Review by taysea — APPROVED** [review]

**Comment by SOjaHPE** [comment]
@halocline  and @taysea  thank you for reviewing, making suggestions and merging the PR! 👍

**Pull request merged**